AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -954,16 +972,17 @@ public class AWSAppSyncClient {
                                       syncConfiguration: syncConfiguration, handlerQueue: queue)
     }
     
-    /// Fetches a query from the server or from the local cache, depending on the current contents of the cache and the specified cache policy.
+    /// Perofmrs a sync operation where a subscription is initiated for real-time updates and a base query or a delta query is used to fetch data from the server.
     ///
     /// - Parameters:
-    ///   - query: The query to fetch.
-    ///   - cachePolicy: A cache policy that specifies when results should be fetched from the server and when data should be loaded from the local cache.
-    ///   - queue: A dispatch queue on which the result handler will be called. Defaults to the main queue.
-    ///   - resultHandler: An optional closure that is called when query results are available or when an error occurs.
-    ///   - result: The result of the fetched query, or `nil` if an error occurred.
-    ///   - error: An error that indicates why the fetch failed, or `nil` if the fetch was succesful.
-    /// - Returns: An object that can be used to cancel an in progress fetch.
+    ///   - baseQuery: The base query to fetch which contains the primary data.
+    ///   - baseQueryResultHandler: Closure that is called when base query results are available or when an error occurs. Every time a sync operation is called, a fetch for the baseQuery from the cache will be done first before initiating any other operations.
+    ///   - subscription: The subscription query which will provide real time updates.
```

Comment:
1. Is it a subscription "query" or a subscription request with callbacks, etc? It's not clear from the docs what the relationship between `subscription` and `deltaQuery` are.
1. The number of parameters in this signature makes me uncomfortable. We have a number of places where we pass GraphQL operations and callbacks; would it make sense to abstract those into a single type that can encapsulate these closely-coupled behaviors?

Reply:
1. Fixed documentation to make it clear.
2. It would definitely be a good idea to move the operations and their completion handlers into a single entity, but due to the dependency we have it is difficult and would require a lot of changes. Will discuss this offline.